.github/workflows/deploy-hnsesb.yml

```diff
@@ -0,0 +1,305 @@
+# This is a multi job workflow to deploy a selected release build or feature build in selected environment
+# No approval require for deploying feature build.
+# Approval required for deploying release build to any environment.
+# Steps are:
+# Job 1:
+#   Validates if it is valid input
+#   Determine target namespace for the input
+#   Determine image tag from the input
+#   Login to openshift and Determine image sha from impage repository 
+# Job 2: Feature build deploy 
+#   Push the image to target repository
+#   Update GitOps repo with sha to deploy using ArgoCD autoSync   
+#   
+# Job 3: Relase build deploy
+#   After deploy request is approved
+#   Push the image to target repository
+#   Update GitOps repo with sha to deploy using ArgoCD autoSync
+
+
+name: deploy-hnsesb
+
+env:
+  OPENSHIFT_SERVER: ${{secrets.C5839F_OPENSHIFT_SERVER}}
+  OPENSHIFT_TOKEN: ${{secrets.C5839F_OPENSHIFT_GITHUB_ACTIONS_SA_TOKEN}}
+  IMAGE_NAME: hnsesb
+  DEFAULT_NAMESPACE: "c5839f-tools"
+  TARGET_NAMESPACE:
+  FEATURE_BUILD_IMAGE_TAG_PREFIX: "FeatureBuild"
+  RELEASE_BUILD_IMAGE_TAG_PREFIX: "Release-1.0"
+  IMAGE_ID:
+
+# Controls when the workflow will run
+on:
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      Environment:
+        type: choice
+        description: Select an environment to deploy
+        options:
+        - dev
+        - test
+        - sandbox
+        - production
+      Build_Type:
+        type: choice
+        description: Select type of build
+        options:
+        - Feature
+        - Release
+      Branch:
+        description: 'Enter feature branch name. For release build deployments this input is ignored.'
+        required: true
+        default: main
+      Build_Run_Number:
+        description: 'Enter feature build run number: '
+        required: true
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  validate:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    
+    outputs:
+      IMAGE_ID: ${{env.IMAGE_ID}}
+      IMAGE_TAG: ${{env.IMAGE_TAG}}
+      TARGET_NAMESPACE: ${{env.TARGET_NAMESPACE}}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+
+      # Find the namespace and image digest from input parameters
+      - name: Compute target namespace
+        id: get-namespace
+        uses: actions/github-script@v5
+        with:
+          script: |
+            if(`${{github.event.inputs.Build_Type}}` == 'Feature'){
```

Comment:
${{github.event.inputs.Environment}} is input by user when running this workflow.
You suggesting to set this value in a local variable and use the variable ? Local variable will look like ${{env.Environment}} in the script.

Reply:
I assume this is JS? I just meant defining a const for `${{github.event.inputs.Environment}}` so that it doesn't need to be repeated throughout the script. It's not that important.

Reply:
updated as per discussion